src/test/java/com/puppycrawl/tools/checkstyle/PropertyCacheFileTest.java

```diff
@@ -90,10 +90,10 @@ public void testCtor() {
     }
 
     @Test
-    public void testInCache() throws IOException {
+    public void testInCache() {
         final Configuration config = new DefaultConfiguration("myName");
-        final String filePath = File.createTempFile("junit", null, temporaryFolder).getPath();
-        final PropertyCacheFile cache = new PropertyCacheFile(config, filePath);
+        final Path filePath = temporaryFolder.toPath().resolve("junit.java");
```

Comment:
how can we make name of file to be randomized ?
in future we plan to make tests to be executed in parallel, so this common name `junit.java` will be a problem.

can we combile it like https://www.javacodegeeks.com/2021/03/creating-temporary-files-with-junit-5.html

please check that files are removed after execution (it was whole point of issue you resolve).

Reply:
> in future we plan to make tests to be executed in parallel, so this common name `junit.java` will be a problem.

I am not sure if this can be considered as a good practice, but we can append UIID. 

```
final String uniqueFileName = "empty_" + UUID.randomUUID() + ".java";
        tempFilePath = Files.createFile(temporaryFolder.toPath().resolve(uniqueFileName)).toFile();
```

Reply:
> please check that files are removed after execution (it was whole point of issue you resolve).

I did build a simple dummy application to test if `@TempDir` works. I didn't see any such directory created using the annotation in temp folder.

Reply:
> can we combile it like https://www.javacodegeeks.com/2021/03/creating-temporary-files-with-junit-5.html

Thank you for the reference. I did help me undersand the issue better. I will soon push the PR based on my understanding so far.

Reply:
> > can we combile it like https://www.javacodegeeks.com/2021/03/creating-temporary-files-with-junit-5.html
> 
> Thank you for the reference. I did help me undersand the issue better. I will soon push the PR based on my understanding so far after your suggestion on the hack related to unique random names.

Reply:
@romani , @rnveach I have resumed my work on this. Please let me know your feedback on updated PR.

Reply:
@relentless-pursuit , please use guid approach everywhere. Each test class will have it own temp folder, inside temp folder we should guarantee unique names.

Reply:
> @relentless-pursuit , please use guid approach everywhere. Each test class will have it own temp folder, inside temp folder we should guarantee unique names.

Done.